cl/opinion_page/forms.py

```diff
@@ -78,3 +95,255 @@ class DocketEntryFilterForm(forms.Form):
+        ),
+    )
+
+    pdf_upload = forms.FileField(
+        label="Opinion PDF",
+        required=True,
+        validators=[FileExtensionValidator(["pdf"])],
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.pk = kwargs.pop("pk", None)
+        super(TennWorkersForm, self).__init__(*args, **kwargs)
+        self.initial["court_str"] = self.pk
+        self.initial["court"] = Court.objects.get(pk=self.pk)
+
+        q_judges = Person.objects.filter(
+            positions__court_id=self.pk, is_alias_of=None
+        ).order_by("name_first")
+        self.fields["lead_author"].queryset = q_judges
+        self.fields["second_judge"].queryset = q_judges
+        self.fields["third_judge"].queryset = q_judges
+
+        if self.pk == "tennworkcompcl":
+            self.fields["cite_reporter"].widget = forms.Select(
+                choices=[("TN WC", "TN WC")], attrs={"class": "form-control"}
+            )
+            del self.fields["second_judge"]
+            del self.fields["third_judge"]
+
+        else:
+            self.fields["cite_reporter"].widget = forms.Select(
+                choices=[("TN WC App.", "TN WC App.")],
+                attrs={"class": "form-control"},
+            )
+
+    def validate_neutral_citation(self):
+        volume = self.cleaned_data["cite_volume"]
+        reporter = self.cleaned_data["cite_reporter"]
+        page = self.cleaned_data["cite_page"]
+
+        c = Citation.objects.filter(
+            volume=volume, reporter=reporter, page=page
+        ).exists()
+        if c:
+            cite = Citation.objects.get(
+                volume=volume, reporter=reporter, page=page
+            )
+            self.add_error(
+                "cite_page",
+                ValidationError(
+                    format_html(
+                        'Citation already in database. See: <a href="%s">%s</a>'
+                        % (cite.get_absolute_url(), cite.cluster.case_name),
+                    )
+                ),
+            )
+        self.cleaned_data["neutral_citation"] = "%s %s %s" % (
+            volume,
+            reporter,
+            page,
+        )
+
+    def verify_unique_judges(self):
+        if self.pk == "tennworkcompapp":
+            judges = [
+                self.cleaned_data["lead_author"],
+                self.cleaned_data["second_judge"],
+                self.cleaned_data["third_judge"],
+            ]
+            flag = len(set(judges)) == len(judges)
+            if not flag:
+                self.add_error(
+                    "lead_author",
+                    ValidationError("Please select each judge only once."),
+                )
+
+    def clean_pdf_upload(self):
+        pdf_data = self.cleaned_data.get("pdf_upload").read()
+        sha_1 = sha1(force_bytes(pdf_data))
+        exists = Opinion.objects.filter(sha1=sha_1).exists()
+        if exists:
+            op = Opinion.objects.get(sha1=sha_1)
```

Comment:
OK, I gotta comment on this pattern b/c you have it twice (so far). This uses queries unnecessarily. The `exists` clause takes a query, and then the `get` essentially repeats it. Also, if there are more than one documents with a SHA1, the filter will pass, but the get will crash. 

Better would be something like:

```
ops = Opinion.objects.filter(sha1=sha_1)
if len(ops) > 0:
    op = ops[0]
```

That runs *one* query, when you do the `len()` call. Note that this doesn't use `count()` either, because that too would add a query.

You also have this pattern starting on line 222. Please fix both.

(Note also, while I'm here, that you should also stick with `sha1` for your variable names here, just to be consistent. `sha1=sha_1` is weird.)

Reply:
I admit sha1 was weird but it was causing a bug with the same named function

Reply:
Suggest: `sha1_hash`.

Reply:
√